test/tap/pack-file-inclusion.js

```diff
@@ -0,0 +1,82 @@
+// verify that pack includes only the files/dirs we want
+var test = require('tap').test
+var common = require('../common-tap')
+var fs = require('graceful-fs')
+var join = require('path').join
+var mkdirp = require('mkdirp')
+var rimraf = require('rimraf')
+var tar = require('tar')
+var zlib = require('zlib')
+
+var pkg = join(__dirname, 'pack-age')
+
+test('setup', function (t) {
+  rimraf.sync(pkg)
+  mkdirp.sync(pkg)
+  fs.writeFileSync(join(pkg, 'package.json'), JSON.stringify({
```

Comment:
You might be interested in looking at `tacks` module, currently used in the various `legacy-*` tests for setting up fixtures like this.